failure in TestTransfer

See https://circleci.com/gh/weaveworks/weave/5402
```
--- FAIL: TestTransfer (0.06s)
	Error Trace:	allocator_test.go:361
	Error:		Not equal: 0x20 (expected)
			        != 0x40 (actual)
```

With
```patch
diff --git a/ipam/allocator_test.go b/ipam/allocator_test.go
index 5df6060..287e4eb 100644
--- a/ipam/allocator_test.go
+++ b/ipam/allocator_test.go
@@ -369,6 +369,12 @@ func TestTransfer(t *testing.T) {
        alloc1.Stop()
 }
 
+func TestTransfer1000(t *testing.T) {
+       for i := 0; i < 1000; i++ {
+               TestTransfer(t)
+       }
+}
+
 func TestFakeRouterSimple(t *testing.T) {
        const (
                cidr = "10.0.1.7/22"
```
and running

    GO15VENDOREXPERIMENT=1 go test -v -run TestTransfer1000 ./ipam

I have managed to get
```
--- FAIL: TestTransfer1000 (0.27s)
	Error Trace:	allocator_test.go:361
	Error:		Not equal: 0x80 (expected)
			        != 0x100 (actual)	
```
on my desktop.

--- a/ipam/allocator_test.go
+++ b/ipam/allocator_test.go
@@ -318,10 +318,12 @@ func TestTransfer(t *testing.T) {
 	_, err = alloc3.Allocate("bar", subnet, returnFalse)
 	require.True(t, err == nil, "Failed to get address")
 
+	// simulation of periodic gossip
 	alloc2.gossip.GossipBroadcast(alloc2.Gossip())
 	router.Flush()
-	alloc2.gossip.GossipBroadcast(alloc3.Gossip())
+	alloc1.gossip.GossipBroadcast(alloc1.Gossip())
 	router.Flush()
+
 	router.RemovePeer(alloc2.ourName)
 	router.RemovePeer(alloc3.ourName)
 	alloc2.Stop()

